### DISCUSSION TITLE
Child windows?

### DISCUSSION THREAD

**Discussion by nanodeath** [tauri-apps/tauri#5785] [Q&A]
Hi there. I'm looking to create little floating helper windows (think like detached tabs in GIMP, or paint.net) and am having some trouble. My primary development environment right now is Linux (Zorin). For now I'm focusing on the first window, a little color palette.

# Defining the Window

I believe I can follow the [static window](https://tauri.app/v1/guides/features/multiwindow#static-window) guide here, and set `visible: false` in the config...but I'm not sure if that'd be too much trouble to open/reopen if the user decides to close the window using the x button. I'm not sure if closing the window makes it invisible or actually gone yet.

# Handling z-index stuff

## If the main window is visible, the smaller windows should also be visible

Can this be done in Tauri or by the OS? If not, maybe the workaround is "when the main window gains focus, cycle through every other window and focus them, then focus on the main window again." Worried that might cause a flicker.

## Always on top...but only vs. the main window

Not sure if this is possible yet. The color palette window should never be behind the main window.

GIMP does this.

# Closing the main window terminates the application

The color palette window shouldn't be allowed to run on its own. Maybe I can solve this by listening to the `WINDOW_CLOSE_REQUESTED` event on the main window and terminating the entire application explicitly. Would be nice if Tauri handled this automatically though; mark the subwindows as "daemon" windows (like how the JVM terminates if only daemon threads remain).

# Shouldn't show up in the taskbar

This one we can definitely do using `skipTaskbar: true` in the config.

**Comment by nanodeath** [comment]
Hmm, it sounds like [`owner_window`](https://github.com/tauri-apps/tauri/blob/eaf0d71779707c69903aba6106322c8908107242/core/tauri/src/window.rs#L448) is exactly what I want, but it's only supported on Windows. That's unfortunate.

**Reply by amrbashir** [reply]
I think you can get the same thing on macOS by using `parent_window`
